book/content/version_control/vcs-git-merge.md

```diff
@@ -0,0 +1,126 @@
+This is a merge conflict: incompatible changes have been made to the same file.
+
+When a merge conflict arises it will be flagged during the merge process.
+Within the files with conflicts the incompatible changes will be marked so you can fix them:
+
+```
+<<<<<<< HEAD
+print('hello world!!!')
+=======
+print('Hello World')
+>>>>>>> master
+```
+`<<<<<<<`: Indicates the start of the lines that had a merge conflict.
+The first set of lines are the lines from the file that you were trying to merge the changes into.
+
+`=======`: Indicates the break point used for comparison.
+Breaks up changes that the user has committed (above) to changes coming from merge (below) to visually see the differences.
+
+`>>>>>>>`: Indicates the end of the lines that had a merge conflict.
+
+You resolve a conflict by editing the file to manually merge the parts of the file that Git had trouble merging.
+This may mean discarding either your changes or someone else's or doing a mix of the two.
+You will also need to delete the `<<<<<<<`, `=======`, and `>>>>>>>` in the file.
+So in this project, the users may decide in favour of one `hello world` over another, or they may decide to replace the conflict with:
+
+```
+print('Hello World!!!')
+```
+
+Once you have fixed the conflicts commit the new version.
+You have now resolved the conflict.
+If, during the process, you need a reminder of which files the conflicts are in you can use `git status` to find out.
+
+If you find there are particularly nasty conflicts and you want to abort the merge you can use:
+```
+git merge --abort
+```
+
+### Good practice
+
+Before you start trying to resolve conflicts make sure you fully understand the changes and how they are incompatible to avoid the risk of making things more tangled.
+Merge conflicts can be intimidating to resolve, especially if you are merging branches that diverged a great many commits ago which may now have many incompatibilities.
+However, it is a good reminder that your previous versions are safe and you can go about fixing this issue without affecting the past versions.
+This is why it is good practice to ** merges other's changes into your work frequently**.
+
+There are tools available to assist in resolving merge conflicts, some are free, some are not.
+Find and familiarise yourself with one that works for you.
+Commonly used merge tools include [KDiff3](http://kdiff3.sourceforge.net/), [Beyond Compare](https://www.scootersoftware.com/), [Meld](http://meldmerge.org/), and [P4Merge](https://www.perforce.com/products/helix-core-apps/merge-diff-tool-p4merge).
+To set a tool as your default do:
+
+```
+git config --global merge.tool name_of_the_tool
+```
+
+and launch it with:
+
+```
+git mergetool
+```
+
+Fundamentally the best way to deal with merge conflicts is to, so far as is possible, try to avoid them in the first place.
```

Comment:
hihi, I very much like this phrasing

```suggestion
Fundamentally, the best way to deal with merge conflicts is, as far as it is possible, to try to avoid them in the first place.
```